pkg/cmd/backuprestore/etcdclientutils.go

```diff
@@ -0,0 +1,138 @@
+package backuprestore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"google.golang.org/grpc"
+	"io"
+	"k8s.io/klog"
+	"os"
+	"time"
+
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/pkg/transport"
+)
+
+func getEtcdClient(endpoints []string) (*clientv3.Client, error) {
+	dialOptions := []grpc.DialOption{
+		grpc.WithBlock(), // block until the underlying connection is up
+	}
+
+	tlsInfo := transport.TLSInfo{
+		CertFile:      os.Getenv("ETCDCTL_CERT"),
+		KeyFile:       os.Getenv("ETCDCTL_KEY"),
+		TrustedCAFile: os.Getenv("ETCDCTL_CACERT"),
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+
+	cfg := &clientv3.Config{
+		DialOptions: dialOptions,
+		Endpoints:   endpoints,
+		DialTimeout: 2 * time.Second,
+		TLS:         tlsConfig,
+	}
+
+	cli, err := clientv3.New(*cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to make etcd client for endpoints %v: %w", endpoints, err)
+	}
+	return cli, nil
+}
+
+func getClusterVersionAndUpgradeInfo(cli *clientv3.Client) (string, bool, error) {
+	upgradeInProgress := false
+	clusterversionKey := "/kubernetes.io/config.openshift.io/clusterversions/version"
```

Comment:
based on the above we will want code to get this data from apiserver vs etcd direct. I understand this was copied form snapshoter.

Reply:
This function is not used in the current backup at all. I will get rid of it.